doc/source/main-documentation/qibolab.rst

```diff
@@ -790,3 +817,29 @@ Compatible and tested with:
 - Xilinx ZCU216
 
 Technically compatible with any board running ``qibosoq``.
+
+Pulse Simulator
+^^^^^^^^^^^^^^^
+
+The simulation controller that is used exclusively by the emulator. It serves primarily to implement the device model using the selected quantum dynamics simulation library (engine), as well as translate and communicate between objects from ``qibolab`` and the selected engine.
+
+Available simulation engines:
+
+- ``qutip``
+
+Currently ``AcquisitionType.DISCRIMINATION`` and   ``AcquisitionType.INTEGRATION`` are supported. Note that for ``AcquistionType.INTEGRATION`` samples are projected onto the I component.
+
+Currently does not support:
+
+- Couplers
+- Flux pulses
+
+.. admonition:: Qibocal compatibility
+
+    The following protocols are currently compatible with the emulator platform (``default_q0``):
+
+    - `rabi_length_sequences`
+    - `t1_sequences`
+    - `t2_sequences`
+    - `allxy`
+    - `standard_rb`
```

Comment:
Based on this list, it seems to me that no routines involving sweepers work. I understand that sweepers involving duration/delays are not very well defined now (this is true for other drivers too), so I would postpone these fixes to 0.2. However, I would expect at least the rabi (amplitude) routine to work, since this involves only a sweeper in the drive amplitude (it doesn't affect the readout) and it is using discrimination (not integration) which is also supported.

I am fine with merging this without sweeper support, however I would prefer raising a `NotImplementedError` under the `sweep` method, instead of having not working code.

Reply:
Rabi (amplitude) routine works I believe, in fact the original runcards were recalibrated by @andrea-pasquale and @sorewachigauyo via `qibocal`.

Reply:
Could you please provide a qibocal runcard with this routine and parameters that work? When I tried yesterday it looked more like noise. Also, according to @andrea-pasquale only the five routines listed above work, which don’t have sweepers.

Reply:
I don't remember if we run `rabi_amplitude`, for sure `rabi_length_sequences` works. I tried this morning again with `rabi_amplitude` but the output is not reasonable. Did you manage to run `rabi_amplitude` @sorewachigauyo ?

Reply:
In that case I was mistaken. Yes I think `rabi_amplitude` in general has an issue, not so much related to `sweep` but I believe there was some missing factor issue that I am not sure we managed to resolve with @sorewachigauyo in the end. It is a subtlety with how the rabi amplitude enters into the model, not so much a `qibocal` related issue. Nonetheless, maybe the easiest way to temporary address this is to add the `NotImplementedError` as @stavros11 suggested and add it once it is resolved. However, the `sweep` method does work in general.

Reply:
Yes, the sweeper itself is not a problem. In particular, the sweeper type is properly implemented (despite part of it being redundant, but correct).

https://github.com/qiboteam/qibolab/blob/aa4e26dd3904dd2b91c3e94bd7f02a80c595003a/src/qibolab/sweeper.py#L38-L40
https://github.com/qiboteam/qibolab/blob/aa4e26dd3904dd2b91c3e94bd7f02a80c595003a/src/qibolab/instruments/emulator/pulse_simulator.py#L402-L406

Reply:
I checked that the pulse parameters were changing between sweeper runs, so the sweeper is indeed working as intended. I also checked the output of `samples` from each sweeper run and they look correct as well. However, the Qibocal output seems to follow the last sweep execution, so it looks like a problem with how the results are being accumulated.

Reply:
The output of the `sweep` method in the emulator has exactly the same format as the `dummy` platform, since that's the only reference I have. If that is different from other platforms, do let me know how it should be.

Reply:
Nope, Qibocal should make use of dummy. To check the results shapes, there are dedicated tests
https://github.com/qiboteam/qibolab/blob/main/tests/test_result_shapes.py
you can try to run them with the emulator as backend (later on, for the emulator we could even run these tests in the CI).

Reply:
I found the problem. It is related to how samples are reshaped before passing it to the results object. It didn't matter for dummy since outputs are random numbers. This issue should be fixed in eefa732de21d8347df34776c58a6d69ac9b43a71, @stavros11 @andrea-pasquale @sorewachigauyo please try to run `rabi_amplitude` or any other sweeper routine in `qibocal` again and let me know if this problem is still not resolved.

Reply:
Thanks @jykhoo1987, I just run `rabi_amplitude` and it seems to work as expected http://login.qrccluster.com:9000/AxNpUTVTT1-Hjw1MNYOtdQ==/
I will test also other protocols using qibocal.